code/modules/client/verbs/looc.dm

```diff
@@ -1,104 +1,86 @@
-            return
-
-    msg = emoji_parse(msg)
-
-    mob.log_talk(raw_msg, LOG_OOC, tag="LOOC")
-
-    var/list/heard = hearers(7, get_top_level_mob(src.mob))
-    for(var/mob/M as() in heard)
-        if(!M.client)
-            continue
-        var/client/C = M.client
-        if (C in GLOB.admins)
-            continue //they are handled after that
-
-        if (isobserver(M))
-            continue //Also handled later.
-
-        if(C.prefs.toggles & CHAT_OOC)
-//            var/display_name = src.key
-//            if(holder)
-//                if(holder.fakekey)
-//                    if(C.holder)
-//                        display_name = "[holder.fakekey]/([src.key])"
-//                else
-//                    display_name = holder.fakekey
-            to_chat(C,"<span class='looc'><span class='prefix'>LOOC:</span> <EM>[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")
-
-    for(var/client/C in GLOB.admins)
-        if(C.prefs.toggles & CHAT_OOC)
-            var/prefix = "(R)LOOC"
-            if (C.mob in heard)
-                prefix = "LOOC"
-            to_chat(C,"<span class='looc'>[ADMIN_FLW(usr)]<span class='prefix'>[prefix]:</span> <EM>[src.key]/[src.mob.name]:</EM> <span class='message'>[msg]</span></span>")
+	set name = "LOOC"
+	set desc = "Local OOC, seen only by those in view."
+	set category = "OOC"
+
+	if(GLOB.say_disabled)    //This is here to try to identify lag problems
+		to_chat(usr, "<span class='danger'> Speech is currently admin-disabled.</span>")
+		return
+
+	if(!mob?.ckey)
+		return
+
+	msg = trim(sanitize(msg), MAX_MESSAGE_LEN)
+	if(!length(msg))
+		return
+
+	var/raw_msg = msg
+
+	if(!(prefs.toggles & CHAT_OOC))
+		to_chat(src, "<span class='danger'>You have OOC (and therefore LOOC) muted.</span>")
+		return
+
+	if(is_banned_from(mob.ckey, "OOC"))
+		to_chat(src, "<span class='danger'>You have been banned from OOC and LOOC.</span>")
+		return
+
+	if(!holder)
+		if(!CONFIG_GET(flag/looc_enabled))
+			to_chat(src, "<span class='danger'>LOOC is disabled.</span>")
+			return
+		if(!GLOB.dooc_allowed && (mob.stat == DEAD))
+			to_chat(usr, "<span class='danger'>LOOC for dead mobs has been turned off.</span>")
+			return
+		if(prefs.muted & MUTE_OOC)
+			to_chat(src, "<span class='danger'>You cannot use LOOC (muted).</span>")
+			return
+		if(handle_spam_prevention(msg, MUTE_OOC))
+			return
+		if(findtext(msg, "byond://"))
+			to_chat(src, "<B>Advertising other servers is not allowed.</B>")
+			log_admin("[key_name(src)] has attempted to advertise in LOOC: [msg]")
+			return
+		if(mob.stat)
+			to_chat(src, "<span class='danger'>You cannot salt in LOOC while unconscious or dead.</span>")
+			return
+		if(isdead(mob))
+			to_chat(src, "<span class='danger'>You cannot use LOOC while ghosting.</span>")
+			return
```

Comment:
@itsmeow non-admin ghosts still can't talk over LOOC, it's just that this allows them to _hear_ LOOC